design/proposals/dbt-delete-merge-schema-branch.md

```diff
@@ -0,0 +1,66 @@
+
+## Required changes
+
+In order to implement the new commands, there's a need to map branches to schemas.
+This is necessary because the branch and the schema might have different names (when adding the `--to-schema <custom name>` flag when creating the schema using the `lakectl dbt create-branch-schema` command) and not use the default behavior (which is the same name for the branch and schema).
+
+### Suggested implementation
+
+The mapping of branches to schemas will be facilitated under the branch's (or more accurately, the commit that this branch is pointing to) metadata property.
+The `schema` key will be used to point to the name of the schema that is configured with the URI of the containing branch: `schema -> <schema name>`.  
+When users start working with lakectl and dbt together they'll also need to map the main/master/production branch they are working with to a schema, otherwise they will be blocked from working with `lakectl dbt` commands.
+This is necessary because, as mentioned in the [Background section](#background), when users first initialize a schema it points to the branch's location, but they didn't take any action to **map the branch to the schema**.
+It might cause a problem if, for example, a user branches out of the main branch (the branch that the target schema's location is pointing to), commits changes with the newly created schema, merges the branches and then wants to merge the schemas. The user wouldn't be able to merge the schemas because the main branch is not aware of the schema associated with it.  
+We can allow users to map a branch to a schema by providing a `lakectl dbt map-branch-schema <branch> <schema>` command. 
+* The functionality of this subcommand is not limited to `lakectl dbt` and can be used in the `lakectl metastore` command suite as well. It's still suggested under the `lakectl dbt` command suite to make it easier for dbt users to integrate lakeFS to their development flow.
+
+## Limitations
+
+* The complete process of merge, i.e. both branch and schema merging, is lacking atomicity. That means that the merge can be partially successful. For example, it's possible that the branch merge was successful but the schema merge was not. This will be resolved in future work.
+
+## Future improvements
+
+### Switch used schema by branch
+
+Make it easier for users to switch the schema they are running their dbt commands over.
+Currently, users need to manually change the schema they want to run their dbt commands over in the profile target's schema property. This means they need to be aware of changes they make to the dbt profile file and not check-in those changes to their version control system later. It also means they need to book the different branch to schema mapping.
+- `lakectl dbt use-schema --branch <branch name>`: Use the provided branch's mapped schema within the terminal that the user uses to run dbt commands.
```

Comment:
I don't think this syntax works in UN*X shells.  The phrasing hints that the command changes an environment variable, but subprocesses cannot change environment variables in the parent.  The best you could do would be e.g.
```sh
`lakectl dbt use-schema --branch <branch_name>`
```
to run the command and interpolate its output into the same CLI used to call it.

Reply:
So the use of it would be something like:
`export LAKEFS_SCHEMA=$(lakectl dbt use-schema --branch <branch_name>)` 
?